### ISSUE TITLE
react-native / react-figma / react-primitives supporting

### ISSUE THREAD

**Issue by ilyalesik** [bernaferrari/FigmaToCode#11]
Proposal: react-native / react-figma / react-primitives supporting

The [react-primitives](https://github.com/lelandrichardson/react-primitives) family have several universal interfaces:

* View
* Text
* StyleSheet
* Platfrom

**Comment by bernaferrari** [comment]
My biggest issue is that I don't know react native. How do I make rectangles, flex, etc in it? If you are willing to help, we can do it. It is not hard, there are just too many details.

**Comment by ilyalesik** [comment]
Sure, I'm very interested. I already made a little [demo](https://github.com/react-figma/code-generator), the proof-of-concept of how it can works. I think that your plugin is most advanced right now, so extending them looks most reasonable.

**Comment by bernaferrari** [comment]
I can make a branch with a wrong conversion and you can point everything
that's wrong, what do you think?

**Comment by ilyalesik** [comment]
@bernaferrari, it's a great idea! You can start a branch; I'll look at the result and start correcting.

**Comment by bernaferrari** [comment]
I have added HTML support to the main branch. Do you think it is something that can replace <div> with <View> or should it be treated as a whole new language?

**Comment by ilyalesik** [comment]
@bernaferrari also styles supporting needed. React Native doesn't support CSS classes; there are their own styles instead: https://reactnative.dev/docs/style.

**Comment by bernaferrari** [comment]
Is there no inline styling? Should we export the layer as component?

**Comment by ilyalesik** [comment]
React Native supports inline styling

**Comment by bernaferrari** [comment]
So, the difference is just div -> View and p to Text? I support JSX already.

**Comment by DaleWebb** [comment]
> So, the difference is just div -> View and p to Text? I support JSX already.

Yes, that's effectively it. Another thing to consider is that the `View` layout only supports flexbox-like layouts. I think your implementation already relies on flexbox, so this probably won't be an issue

**Comment by TheUltDev** [comment]
For those using React Native, I'm working on a plugin that's specifically for Figma -> React Native:
https://github.com/kat-tax/figaro

Thank you @bernaferrari, your project was useful for the flexbox/sizing related code. I will be adding attribution.
